### ISSUE TITLE
Cannot remote debug Java 9+

### ISSUE THREAD

**Issue by neelkamath** [gradle/gradle#13118]
Using `gradle test --debug-jvm` doesn't work for Java 9+ remote debugging. This affects Java, Kotlin, and probably every JVM language targeting JVM 9+ bytecode.

### Expected Behavior

When running `gradle --debug-jvm` in a Java 9+ project, the argument Gradle passes to the JVM is supposed to be `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=*:5005` ([reference](https://dev-aux.com/java/java-local-and-remote-jvm-debugging-jdk-8-and-later)).

### Current Behavior
When running `gradle --debug-jvm`, the `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005` argument is passed by Gradle to the JVM regardless of the JVM target being used. This causes it to only work for JVM 8- bytecode. This behavior can be seen in the [docs](https://docs.gradle.org/current/userguide/build_environment.html):

> **`org.gradle.debug=(true,false)`**
When set to `true`, Gradle will run the build with remote debugging enabled, listening on port 5005. Note that this is the equivalent of adding `-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005` to the JVM command line and will suspend the virtual machine until a debugger is attached. Default is `false`.

### Context

This issue has affected me because I cannot debug my containerized Kotlin server, which targets JVM 13 bytecode.

### Steps to Reproduce 

https://github.com/neelkamath/gradle-debug-bug

### Your Environment

Build scan URL: https://scans.gradle.com/s/7dyy6iwaeqhli

**Comment by neelkamath** [comment]
I found a workaround. Just modify the `test` task in your `build.gradle.kts`:
```kotlin
tasks {
    withType<Test> {
        useJUnitPlatform()
        // Workaround for remote debugging a JVM 9+ target (see https://github.com/gradle/gradle/issues/13118).
        jvmArgs = listOf("-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=*:5005")
    }
}
```

A caveat is that it causes Gradle to ask for the debugger to get attached every time you run `gradle test`. However, it doesn't wait indefinitely in case you don't attach one, and will continue with the tests.

**Comment by mlopatkin** [comment]
Thank you for your interest in Gradle!

This issue will be closed as a duplicate of #7496. Please add your +1's to that issue.
